system/jlib/jbuff.cpp

```diff
@@ -243,13 +254,21 @@ void MemoryBuffer::_realloc(size32_t newLen)
                 newMax = FIRST_CHUNK_SIZE;
             while (newLen > newMax)
             {
-                newMax += newMax;
+                size32_t newMaxTmp = newMax + newMax;
+                if (newMaxTmp < newMax)
+                    RaiseOutOfMemException(-11, newMax, newMax, false, "Exceeded maximum size");
```

Comment:
This error will says wanted : X had X, which will still be a bit wrong/confusing.
And one below will show overflowed small unsigned32 newMaxTmp for wanted.

If you common up into and inline check that throws, it will correct and be clearer I think.